everyvoice/tests/test_model.py

```diff
@@ -251,3 +253,185 @@ def test_model_is_not_a_vocoder(self):
+                logger=False,
+                max_epochs=1,
+                limit_train_batches=1,
+                limit_val_batches=1,
+                callbacks=[ModelCheckpoint(dirpath=tmpdir_str, every_n_train_steps=1)],
+            )
+            trainer.strategy.connect(model)
+            ckpt_fn = tmpdir_str + "/checkpoint.ckpt"
+            trainer.save_checkpoint(ckpt_fn)
+            m = torch.load(ckpt_fn)
+            self.assertIn("model_info", m.keys())
+            m["model_info"]["name"] = "BAD_TYPE"
+            torch.save(m, ckpt_fn)
+            m = torch.load(ckpt_fn)
+            self.assertIn("model_info", m.keys())
+            self.assertEqual(m["model_info"]["name"], "BAD_TYPE")
+            # self.assertEqual(m["model_info"]["version"], "1.0")
+            with self.assertRaisesRegex(
+                TypeError,
+                r"Wrong model type \(BAD_TYPE\), we are expecting a FastSpeech2",
+            ):
+                FastSpeech2.load_from_checkpoint(ckpt_fn)
+
+    def test_wrong_model_version(self):
+        """
+        Detecting wrong model version number in checkpoint.
+        """
+        from pytorch_lightning import Trainer
+        from pytorch_lightning.callbacks import ModelCheckpoint
+
+        with tempfile.TemporaryDirectory() as tmpdir_str:
+            model = FastSpeech2(
+                FastSpeech2Config.load_config_from_path(
+                    self.config_dir / f"{TEXT_TO_SPEC_CONFIG_FILENAME_PREFIX}.yaml"
+                ),
+                stats=Stats(
+                    pitch=StatsInfo(
+                        min=0, max=1, std=2, mean=3, norm_min=4, norm_max=5
+                    ),
+                    energy=StatsInfo(
+                        min=7, max=8, std=9, mean=10, norm_min=11, norm_max=12
+                    ),
+                ),
+                lang2id={"foo": 0, "bar": 1},
+                speaker2id={"baz": 0, "qux": 1},
+            )
+            BAD_VERSION = "BAD_VERSION"
+            model._VERSION = BAD_VERSION
+            trainer = Trainer(
+                default_root_dir=tmpdir_str,
+                enable_progress_bar=False,
+                logger=False,
+                max_epochs=1,
+                limit_train_batches=1,
+                limit_val_batches=1,
+                callbacks=[ModelCheckpoint(dirpath=tmpdir_str, every_n_train_steps=1)],
+            )
+            trainer.strategy.connect(model)
+            ckpt_fn = tmpdir_str + "/checkpoint.ckpt"
+            trainer.save_checkpoint(ckpt_fn)
+            m = torch.load(ckpt_fn)
+            self.assertIn("model_info", m.keys())
+            self.assertEqual(m["model_info"]["name"], FastSpeech2.__name__)
+            self.assertEqual(m["model_info"]["version"], BAD_VERSION)
+            with self.assertRaisesRegex(
+                ValueError,
+                r"Your model was created with a newer version of EveryVoice, please update your software.",
```

Comment:
I'm surprised that "BAD_VERSION" causes an error message that says "your model was created with a newer version of EveryVoice" - isn't "BAD_VERSION" just an unknown version? I would rather see messaging that indicates whether the provided version is newer/older or unknown.